### ISSUE TITLE
Should not lint on locally unused @exported data objects

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#32]
## box.linters version

0.0.0.9006

## Source code to lint

```R
#' @export
public_data <- c(1, 3)
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Expected result

Should not lint on unused `public_data`.
